src/main/java/com/pingcap/tikv/types/TimestampType.java

```diff
@@ -77,12 +77,11 @@ public void decode(CodecDataInput cdi, Row row, int pos) {
       if (localDateTime == null) {
         row.setNull(pos);
       }
-      Timestamp timestamp = Timestamp.valueOf(localDateTime);
+      Timestamp timestamp = Timestamp.from(ZonedDateTime.of(localDateTime, ZoneId.of("UTC")).toInstant());
```

Comment:
Please confirm if ZoneId is immutable and can be replaced by final constant. If so, avoid create object please here since this will be called per row and a lot of object creation involved.

Reply:
DONE.